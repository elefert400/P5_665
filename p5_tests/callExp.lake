int foo(){
    return 1;
}
bool bar(){
    return false;
}
void func(int x, bool b){}
int main(){
    func(foo() + 1, bar());
}