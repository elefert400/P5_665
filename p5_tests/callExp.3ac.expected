[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN foo LOCALS]
[END foo LOCALS]
enter foo
setout 1 1
goto lbl_0
lbl_0: leave foo
[BEGIN bar LOCALS]
[END bar LOCALS]
enter bar
setout 1 0
goto lbl_1
lbl_1: leave bar
[BEGIN func LOCALS]
x (formal)
b (formal)
[END func LOCALS]
enter func
getin 1 x
getin 2 b
lbl_2: leave func
[BEGIN main LOCALS]
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
[END main LOCALS]
enter main
call foo
getout 1 tmp0
tmp1 := tmp0 ADD 1
setin 1 tmp1
call bar
getout 1 tmp2
setin 2 tmp2
call func
lbl_3: EXIT